sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -0,0 +1,853 @@
+        """
+        entry_ele = await self._get_entity_element(topic_name, **kwargs)
+        entry = TopicDescriptionEntry.deserialize(entry_ele)
+        if not entry.content:
+            raise ResourceNotFoundError("Topic {} does not exist".format(topic_name))
+        topic_description = TopicRuntimeInfo._from_internal_entity(entry.content.topic_description)
+        topic_description.name = topic_name
+        return topic_description
+
+    async def create_topic(self, topic: Union[str, TopicDescription], **kwargs) -> TopicDescription:
+        """Create a topic.
+
+        :param Union[str, ~azure.servicebus.management.TopicDescription] topic: The topic name or a `TopicDescription`
+         instance. When it's a str, it will be the name of the created topic. Other properties of the created topic
+         will have default values decided by the ServiceBus.
+         Use a `TopicDescription` if you want to set queue properties other than the queue name.
+        :rtype: ~azure.servicebus.management.TopicDescription
+        """
+        try:
+            topic_name = topic.name  # type: ignore
+            to_create = topic._to_internal_entity()  # type: ignore
+        except AttributeError:
+            topic_name = topic  # type: ignore
+            to_create = InternalTopicDescription()  # Use an empty topic description.
+
+        create_entity_body = CreateTopicBody(
+            content=CreateTopicBodyContent(
+                topic_description=to_create,  # type: ignore
+            )
+        )
+        request_body = create_entity_body.serialize(is_xml=True)
+        with _handle_response_error():
+            entry_ele = cast(
+                ElementTree,
+                await self._impl.entity.put(
+                    topic_name,  # type: ignore
+                    request_body, api_version=constants.API_VERSION, **kwargs)
+            )
+        entry = TopicDescriptionEntry.deserialize(entry_ele)
+        result = TopicDescription._from_internal_entity(entry.content.topic_description)
+        result.name = topic_name
+        return result
+
+    async def update_topic(self, topic_description: TopicDescription, **kwargs) -> None:
+        """Update a topic.
+
+        Before calling this method, you should use `get_topic` to get a `TopicDescription` instance, then use the
+        keyword arguments to update the properties you want to update.
+        Only a portion of properties can be updated.
+        Refer to https://docs.microsoft.com/en-us/rest/api/servicebus/update-topic.
+
+        :param ~azure.servicebus.management.TopicDescription topic_description: The topic to be updated.
+        :keyword timedelta default_message_time_to_live: The value you want to update to.
+        :keyword timedelta duplicate_detection_history_time_window: The value you want to update to.
+        :rtype: None
+        """
+
+        if not isinstance(topic_description, TopicDescription):
```

Comment:
What was our justification for doing this here and not elsewhere?  This could be reasonable, since I notice you're consistent across all update calls (just chose this one arbitrarily) but wanted to understand the why behind this choice for my own education.

Reply:
Removed all such checks in update_xxx. I don't think we need them. Good catch.